src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppExtensions.cs

```diff
@@ -31,4 +43,266 @@ public static IDistributedApplicationBuilder AddAzureContainerAppsInfrastructure
+    {
+        builder.AddAzureContainerAppsInfrastructure();
+
+        // Only support one temporarily until we can support multiple environments
+        // and allowing each container app to be explicit about which environment it uses
+        if (builder.Resources.OfType<AzureContainerAppEnvironmentResource>().Any())
+        {
+            throw new InvalidOperationException("Only one container app environment is supported at this time. Please remove the existing container app environment.");
+        }
+
+        var containerAppEnvResource = new AzureContainerAppEnvironmentResource(name, static infra =>
+        {
+            var principalId = new ProvisioningParameter("principalId", typeof(string));
+
+            infra.Add(principalId);
+
+            var tags = new ProvisioningParameter("tags", typeof(object))
+            {
+                Value = new BicepDictionary<string>()
+            };
+
+            infra.Add(tags);
+
+            var identity = new UserAssignedIdentity("mi")
+            {
+                Tags = tags
+            };
+
+            infra.Add(identity);
+
+            var containerRegistry = new ContainerRegistryService("acr")
+            {
+                Sku = new() { Name = ContainerRegistrySkuName.Basic },
+                Tags = tags
+            };
+
+            infra.Add(containerRegistry);
+
+            var pullRa = containerRegistry.CreateRoleAssignment(ContainerRegistryBuiltInRole.AcrPull, identity);
+
+            // There's a bug in the CDK, see https://github.com/Azure/azure-sdk-for-net/issues/47265
+            pullRa.Name = BicepFunction.CreateGuid(containerRegistry.Id, identity.Id, pullRa.RoleDefinitionId);
+            infra.Add(pullRa);
+
+            var pushRa = containerRegistry.CreateRoleAssignment(ContainerRegistryBuiltInRole.AcrPush, identity);
+            pushRa.Name = BicepFunction.CreateGuid(containerRegistry.Id, identity.Id, pushRa.RoleDefinitionId);
+            infra.Add(pushRa);
+
+            var laWorkspace = new OperationalInsightsWorkspace("law")
+            {
+                Sku = new() { Name = OperationalInsightsWorkspaceSkuName.PerGB2018 },
+                Tags = tags
+            };
+
+            infra.Add(laWorkspace);
+
+            var containerAppEnvironment = new ContainerAppManagedEnvironment("cae")
+            {
+                WorkloadProfiles = [
+                    new ContainerAppWorkloadProfile()
+                    {
+                        WorkloadProfileType = "Consumption",
+                        Name = "consumption"
+                    }
+                ],
+                AppLogsConfiguration = new()
+                {
+                    Destination = "log-analytics",
+                    LogAnalyticsConfiguration = new()
+                    {
+                        CustomerId = laWorkspace.CustomerId,
+                        SharedKey = laWorkspace.GetKeys().PrimarySharedKey
```

Comment:
I know this is how azd does it, but is there a way to not use secrets here? For the future - obviously we won't change this in this PR. But should we consider logging a bug to use MSI/Entra here?

Reply:
I think that would be an ACA issue